File: app/src/main/java/com/example/learn/testme/Subject.java
package com.example.learn.testme;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.learn.testme.Common.common;
import com.example.learn.testme.Interface.ItemClickLIstener;
import com.example.learn.testme.Model.Subjecte;
import com.example.learn.testme.ViewHolder.SubjecteViewHolder;
import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;


public class Subject extends Fragment {



    View myfragment;

    RecyclerView listSubjecte;
    RecyclerView.LayoutManager layoutManager;
    FirebaseRecyclerAdapter<Subjecte,SubjecteViewHolder> adapter;

    FirebaseDatabase database;
    DatabaseReference subjecteDB;


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        database = FirebaseDatabase.getInstance();
        subjecteDB = database.getReference("Subjecte");

    }

    public static Subject newInstance(){
        Subject subject = new Subject();
        return subject;

    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        myfragment = inflater.inflate(R.layout.fragment_subject,container,false);

        listSubjecte = (RecyclerView)myfragment.findViewById(R.id.listSubjecte);
        listSubjecte.setHasFixedSize(true);;
        layoutManager = new LinearLayoutManager(container.getContext());
        listSubjecte.setLayoutManager(layoutManager);

        
        loadSubject();

       
        return myfragment;
    }



    //load subject use category id is index for load

    private void loadSubject() {

        adapter = new FirebaseRecyclerAdapter<Subjecte, SubjecteViewHolder>(
                Subjecte.class,
                R.layout.subjecte_layout,
                SubjecteViewHolder.class,
                subjecteDB.orderByChild("Category_id").equalTo(common.categoryId)

        ){
            @Override
            protected void populateViewHolder(SubjecteViewHolder viewHolder, final Subjecte model, int position) {
                  viewHolder.Subjecte_Nmae.setText(model.getSubjecte_Name());
                  viewHolder.Subjecte_Detail.setText(model.getSubjecte_Deteil());

                  viewHolder.setItemClickListener(new ItemClickLIstener() {
                      @Override
                      public void onClick(View view, int position, boolean isLongClick) {



                     common.Subjecte_Id =  adapter.getRef(position).getKey();
                     Log.e("Test Click subjecte >> ", common.Subjecte_Id);
                     common.Subjecte_Deteil = model.getSubjecte_Deteil();
                     Log.e("Test Click Subjecte Deteil",common.Subjecte_Deteil);
                      common.Subjecte_Name = model.getSubjecte_Name();
                     Intent startgame = new Intent(getActivity(),Start.class);
                     startActivity(startgame);
                      }
                  });

            }


        };
        adapter.notifyDataSetChanged();
        listSubjecte.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/learn/testme/Subject.kt
package com.example.learn.testme

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.learn.testme.Model.Subjecte
import com.example.learn.testme.ViewHolder.SubjecteViewHolder
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase


class Subject : androidx.fragment.app.Fragment() {
    var myfragment: View? = null
    var listSubjecte: androidx.recyclerview.widget.RecyclerView? = null
    var layoutManager: androidx.recyclerview.widget.RecyclerView.LayoutManager? = null
    var adapter: FirebaseRecyclerAdapter<Subjecte?, SubjecteViewHolder>? = null
    var database: FirebaseDatabase? = null
    var subjecteDB: DatabaseReference? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        database = FirebaseDatabase.getInstance()
        subjecteDB = database!!.getReference("Subjecte")
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        myfragment = inflater.inflate(R.layout.fragment_subject, container, false)
        listSubjecte = myfragment!!.findViewById<View>(R.id.listSubjecte) as androidx.recyclerview.widget.RecyclerView
        listSubjecte!!.setHasFixedSize(true)
        layoutManager = androidx.recyclerview.widget.LinearLayoutManager(container!!.context)
        listSubjecte!!.layoutManager = layoutManager
        loadSubject()
        return myfragment
    }

    //load subject use category id is index for load
    private fun loadSubject() {
//        adapter = object : FirebaseRecyclerAdapter<Subjecte, SubjecteViewHolder>(
//                Subjecte,
//                R.layout.item_subjecte,
//                SubjecteViewHolder::class.java,
//                subjecteDB!!.orderByChild("Category_id").equalTo(common.categoryId)
//        ) {
//            override fun populateViewHolder(viewHolder: SubjecteViewHolder, model: Subjecte, position: Int) {
//                viewHolder.Subjecte_Nmae.text = model.subjecte_Name
//                viewHolder.Subjecte_Detail.text = model.subjecte_Deteil
//                viewHolder.setItemClickListener { view, position, isLongClick ->
//                    common.Subjecte_Id = adapter!!.getRef(position).key
//                    Log.e("Test Click subjecte >> ", common.Subjecte_Id)
//                    common.Subjecte_Deteil = model.subjecte_Deteil
//                    Log.e("Test Click Subjecte Deteil", common.Subjecte_Deteil)
//                    common.Subjecte_Name = model.subjecte_Name
//                    val startgame = Intent(activity, Start::class.java)
//                    startActivity(startgame)
//                }
//            }
//        }
//        adapter.notifyDataSetChanged()
        listSubjecte!!.adapter = adapter
    }

    companion object {
        @JvmStatic
        fun newInstance(): Subject {
            return Subject()
        }
    }
}